### PR TITLE
More relaxed inheritance mapping.

### PR DESCRIPTION
**Pull request by meistermeier**
With this PR it will be possible to map concrete classes coming from queries of abstract nodes without an annotation and as a consequence parent label.

### CODE REVIEW

File: src/test/java/org/springframework/data/neo4j/integration/imperative/InheritanceMappingIT.java [hunk 1]
```diff
@@ -129,10 +132,31 @@ void resultCollectionShouldHaveCorrectTypes(@Autowired TerritoryRepository repos
...
+			Transaction transaction = session.beginTransaction();
+			transaction.run("CREATE (:Cat{name:'a'})");
+			transaction.run("CREATE (:Cat{name:'a'})");
+			transaction.run("CREATE (:Cat{name:'a'})");
+			transaction.run("CREATE (:Dog{name:'a'})");
+			transaction.commit();
+		}
+
+		List<AbstractNode> pets = petsRepository.findPets("a");
+		assertThat(pets).hasSize(4);
```
Comment (michael-simons) [inline src/test/java/org/springframework/data/neo4j/integration/imperative/InheritanceMappingIT.java:148]: Can we make sure those are different?

File: src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jPersistentEntity.java [hunk 2]
```diff
@@ -42,5 +42,17 @@
 	 */
 	Optional<Neo4jPersistentProperty> getDynamicLabelsProperty();
 
+	/**
+	 * Determines if the entity is annotated with {@link org.springframework.data.neo4j.core.schema.RelationshipProperties}
+	 *
+	 * @return true if this is a relationship properties class, otherwise false.
+	 */
```
Comment (michael-simons) [inline src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jPersistentEntity.java:49]: Unrelated but nice change :)

File: src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jPersistentEntity.java [hunk 3]
```diff
@@ -42,5 +42,17 @@
...
 	boolean isRelationshipPropertiesEntity();
+
+	/**
+	 * Determines if this entity is annotated with {@link org.springframework.data.neo4j.core.schema.Node}.
+	 *
+	 * @return true if this is an intentional entity, otherwise false.
+	 */
+	boolean isExplicitEntity();
```
Comment (michael-simons) [inline src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jPersistentEntity.java:57]: Can we please not add this? Instead do it internally the way it was before inside the mapping context?

File: src/test/java/org/springframework/data/neo4j/integration/shared/common/AbstractNode.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.neo4j.core.schema.GeneratedValue;
+import org.springframework.data.neo4j.core.schema.Id;
+
+/**
+ * @author Gerrit Meier
+ */
+public abstract class AbstractNode {
```
Comment (michael-simons) [inline src/test/java/org/springframework/data/neo4j/integration/shared/common/AbstractNode.java:26]: `AbstractAnimal` or make cat and dogs ConcreteNode1 and 2.

**Review by michael-simons — CHANGES_REQUESTED** [review]
Only high level complains.

**Review by michael-simons — APPROVED** [review]
Looks great now.

**Comment by meistermeier** [comment]
Merged with 01fa71994 (6.0.x) and c9fcb1433 (master)

**Pull request closed without merging**